src/cli/test/parse-flags.spec.ts

```diff
@@ -30,556 +20,216 @@ describe('parseFlags', () => {
+   * will cause these tests to start failing.
    */
   describe.each(BOOLEAN_CLI_ARGS)('should parse boolean flag %s', (cliArg) => {
     it('should parse arg', () => {
-      const flags = parseFlags([`--${cliArg}`], sys);
+      const flags = parseFlags([`--${cliArg}`]);
       expect(flags.knownArgs).toEqual([`--${cliArg}`]);
       expect(flags[cliArg]).toBe(true);
     });
 
     it(`should parse --no${cliArg}`, () => {
       const negativeFlag = '--no' + cliArg.charAt(0).toUpperCase() + cliArg.slice(1);
-      const flags = parseFlags([negativeFlag], sys);
+      const flags = parseFlags([negativeFlag]);
       expect(flags.knownArgs).toEqual([negativeFlag]);
       expect(flags[cliArg]).toBe(false);
     });
-  });
 
-  it.each(STRING_CLI_ARGS)('should parse string flag %s', (cliArg) => {
-    const flags = parseFlags([`--${cliArg}`, 'test-value'], sys);
-    expect(flags.knownArgs).toEqual([`--${cliArg}`, 'test-value']);
-    expect(flags.unknownArgs).toEqual([]);
-    expect(flags[cliArg]).toBe('test-value');
+    it(`should override --${cliArg} with --no${cliArg}`, () => {
+      const negativeFlag = '--no' + cliArg.charAt(0).toUpperCase() + cliArg.slice(1);
+      const flags = parseFlags([`--${cliArg}`, negativeFlag]);
+      expect(flags.knownArgs).toEqual([`--${cliArg}`, negativeFlag]);
+      expect(flags[cliArg]).toBe(false);
+    });
+
+    it('should set value to null if not present', () => {
+      const flags = parseFlags([]);
+      expect(flags.knownArgs).toEqual([]);
+      expect(flags[cliArg]).toBe(null);
+    });
   });
 
-  it.each(STRING_CLI_ARGS)('should parse string flag --%s=value', (cliArg) => {
-    const flags = parseFlags([`--${cliArg}=path/to/file.js`], sys);
-    expect(flags.knownArgs).toEqual([`--${cliArg}`, 'path/to/file.js']);
-    expect(flags.unknownArgs).toEqual([]);
-    expect(flags[cliArg]).toBe('path/to/file.js');
+  describe.each(STRING_CLI_ARGS)('should parse string flag %s', (cliArg) => {
+    it(`should parse "--${cliArg} value"`, () => {
+      const flags = parseFlags([`--${cliArg}`, 'test-value']);
+      expect(flags.knownArgs).toEqual([`--${cliArg}`, 'test-value']);
+      expect(flags.unknownArgs).toEqual([]);
+      expect(flags[cliArg]).toBe('test-value');
+    });
+
+    it(`should parse "--${cliArg}=value"`, () => {
+      const flags = parseFlags([`--${cliArg}=path/to/file.js`]);
+      expect(flags.knownArgs).toEqual([`--${cliArg}`, 'path/to/file.js']);
+      expect(flags.unknownArgs).toEqual([]);
+      expect(flags[cliArg]).toBe('path/to/file.js');
+    });
+
+    it(`should parse "--${toDashCase(cliArg)} value"`, () => {
+      const flags = parseFlags([`--${toDashCase(cliArg)}`, 'path/to/file.js']);
+      expect(flags.knownArgs).toEqual([`--${toDashCase(cliArg)}`, 'path/to/file.js']);
+      expect(flags.unknownArgs).toEqual([]);
+      expect(flags[cliArg]).toBe('path/to/file.js');
+    });
+
+    it(`should parse "--${toDashCase(cliArg)}=value"`, () => {
```

Comment:
I added two more tests for the string arguments here, basically testing that `--foo-bar baz` and `--foo-bar=baz` will be parsed correctly